python/dolfinx/fem/__init__.py

```diff
@@ -13,6 +13,15 @@
 from dolfinx.cpp.fem import transpose_dofmap  # noqa
 from dolfinx.cpp.fem import IntegralType
 from dolfinx.cpp.fem import create_sparsity_pattern as _create_sparsity_pattern
+from dolfinx.fem.dirichletbc import (DirichletBC, dirichletbc, bcs_by_block,
```

Comment:
Should we remove `DirichletBC` and `Form` from import, as they cannot be called without
```python
     formcls = type("DirichletBC", (DirichletBC, bctype), {})
```
and 
```python
     formcls = type("Form", (Form, ftype), {})
```
as the class Form and DirichletBC has no initial base class to inherit from, and call `super()` with.

Reply:
I'm not sure - we still use it with type hints and we want to document the methods.

The analogy to NumPy is `ndarray`. `ndarray` is the 'main' NumPy class, but is created used factory functions.

Reply:
It would be good to signpost to users that these classes are now special metaclasses, either through the name e.g. appending Meta, or improve the docstrings.